starter-code/client/src/app/components/single-entry/single-entry.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit } from '@angular/core';
+import { JournalService } from '../../services/journal.service';
+import { ActivatedRoute } from '@angular/router'
+
+@Component({
+  selector: 'app-single-entry',
+  templateUrl: './single-entry.component.html',
+  styleUrls: ['./single-entry.component.css']
+})
+export class SingleEntryComponent implements OnInit {
+
+  journal: Object;
+
+  journalId: any;
+
+  constructor(private journalService: JournalService, private route: ActivatedRoute) { }
+
+  ngOnInit() {
+    this.route.params
+      .subscribe((params) => {
+        this.journalId = String(params.id);
+      })
+    this.journalService.getSingleJournal(this.journalId).then(journal => {
```

Comment:
Care!
 Line 21 is asynchronous and you are using `this.journalId` on line 23 where can be undefined since is asynchronous.